src/node-opendds.cpp

```diff
@@ -290,14 +316,271 @@ namespace {
+        fci.GetReturnValue().SetUndefined();
+        return;
+      }
+      Registered_Data_Types->register_type(dp, *topic_type, ts);
+    }
+    const OpenDDS::DCPS::V8TypeConverter* const tc =
+      dynamic_cast<const OpenDDS::DCPS::V8TypeConverter*>(ts);
+    if (!tc) {
+      Nan::ThrowError("TypeSupport was not built with support for V8.");
+      fci.GetReturnValue().SetUndefined();
+      return;
+    }
+
+    DDS::Topic_var topic =
+      dp->create_topic(*topic_name, *topic_type, TOPIC_QOS_DEFAULT, 0, 0);
+
+    Local<Object> qos_js;
+    if (fci.Length() > 2 && fci[2]->IsObject()) {
+      qos_js = fci[2]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
+    }
+
+    DDS::PublisherQos pub_qos;
+    dp->get_default_publisher_qos(pub_qos);
+    Nan::MaybeLocal<String> pubqos_str = Nan::New<String>("PublisherQos");
+    const Local<String> pubqos_lstr = pubqos_str.ToLocalChecked();
+    if (*qos_js && qos_js->Has(pubqos_lstr)) {
+      const Local<Value> pubqos_lv = qos_js->Get(pubqos_lstr);
+      if (pubqos_lv->IsObject()) {
+        try {
+          convertQos(pub_qos, pubqos_lv->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
+        } catch (const std::runtime_error& e) {
+          Nan::ThrowError(e.what());
+          fci.GetReturnValue().SetUndefined();
+          return;
+        }
+      }
+    }
+
+    const DDS::Publisher_var pub = dp->create_publisher(pub_qos, 0, 0);
+    if (!pub) {
+      Nan::ThrowError("couldn't create Publisher");
+      fci.GetReturnValue().SetUndefined();
+      return;
+    }
+
+    DDS::DataWriterQos dw_qos;
+    pub->get_default_datawriter_qos(dw_qos);
+    const Local<String> dwqos_lstr = Nan::New<String>("DataWriterQos").ToLocalChecked();
+    if (*qos_js && qos_js->Has(dwqos_lstr)) {
+      const Local<Value> dwqos_lv = qos_js->Get(dwqos_lstr);
+      if (dwqos_lv->IsObject()) {
+        try {
+          convertQos(dw_qos, dwqos_lv->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
+        } catch (const std::runtime_error& e) {
+          Nan::ThrowError(e.what());
+          fci.GetReturnValue().SetUndefined();
+          return;
+        }
+      }
+    }
+
+    DDS::DataWriter_var dw = pub->create_datawriter(topic, dw_qos, 0, 0);
+    if (!dw) {
+      Nan::ThrowError("couldn't create DataWriter");
+      fci.GetReturnValue().SetUndefined();
+      return;
+    }
+
+    const Local<ObjectTemplate> ot = Nan::New<ObjectTemplate>();
+    ot->SetInternalFieldCount(2);
+    Nan::SetMethod(ot, "register_instance", register_instance);
+    Nan::SetMethod(ot, "write", write);
+    Nan::SetMethod(ot, "unregister_instance", unregister_instance);
+    Nan::SetMethod(ot, "dispose", dispose);
+    const Local<Object> obj = ot->NewInstance();
+    Nan::SetInternalFieldPointer(obj, 0, dw._retn());
+    Nan::SetInternalFieldPointer(obj, 1, const_cast<OpenDDS::DCPS::V8TypeConverter*>(reinterpret_cast<const OpenDDS::DCPS::V8TypeConverter*>(tc)));
```

Comment:
Is the `reinterpret_cast` needed?  The type of `tc` appears to already be `V8TypeConverter`.

Reply:
I couldn't figure out how to const_cast away the 2nd const in "const OpenDDS::DCPS::V8TypeConverter* const", though const_cast does seem to work for the first one. So I used reinterpret_cast for the second and then const_cast for the first.

Reply:
Remove the consts from lines 344,345 and then you will not need either cast.

Reply:
We discussed and we think we have a better alternative.

Reply:
;) .. drive by peer review

Reply:
Thanks for looking at it.  It's a classic case of our code needing to interact with a 3rd-party API that is non-const, so just a `const_cast` seems to be fine.

Reply:
Not sure why the simple const_cast wasn't working for me when I first encountered the issue, but it works now so that's what we'll use!